examples/ConnectNoEncryption/ConnectNoEncryption.ino

```diff
@@ -91,61 +92,92 @@ void printWifiData() {
   byte mac[6];
   WiFi.macAddress(mac);
   Serial.print("MAC address: ");
-  Serial.print(mac[5], HEX);
-  Serial.print(":");
-  Serial.print(mac[4], HEX);
-  Serial.print(":");
-  Serial.print(mac[3], HEX);
-  Serial.print(":");
-  Serial.print(mac[2], HEX);
-  Serial.print(":");
-  Serial.print(mac[1], HEX);
-  Serial.print(":");
-  Serial.println(mac[0], HEX);
+  for (uint8_t i = 0; i < 6; i++) {
+    if (mac[i] < 0x10) {
+      Serial.print("0");
+    }
+    Serial.print(mac[i], HEX);
+    if (i != 5) {
+      Serial.print(":");
+    }
+    else {
+      Serial.println();
+    }
+  }
 
   // print your subnet mask:
   IPAddress subnet = WiFi.subnetMask();
   Serial.print("NetMask: ");
   Serial.println(subnet);
 
-  // print your gateway address:
-  IPAddress gateway = WiFi.gatewayIP();
-  Serial.print("Gateway: ");
-  Serial.println(gateway);
+  // print your gateway IP address:
+  IPAddress gatewayIP = WiFi.gatewayIP();
+  Serial.print("Gateway IP: ");
+  Serial.println(gatewayIP);
 }
 
 void printCurrentNet() {
   // print the SSID of the network you're attached to:
   Serial.print("SSID: ");
   Serial.println(WiFi.SSID());
 
-  // print the MAC address of the router you're attached to:
+  // print the BSSID which is the MAC address of the router you're attached to:
   byte bssid[6];
   WiFi.BSSID(bssid);
   Serial.print("BSSID: ");
-  Serial.print(bssid[5], HEX);
-  Serial.print(":");
-  Serial.print(bssid[4], HEX);
-  Serial.print(":");
-  Serial.print(bssid[3], HEX);
-  Serial.print(":");
-  Serial.print(bssid[2], HEX);
-  Serial.print(":");
-  Serial.print(bssid[1], HEX);
-  Serial.print(":");
-  Serial.println(bssid[0], HEX);
+  for (uint8_t i = 0; i < 6; i++) {
+    if (bssid[i] < 0x10) {
+      Serial.print("0");
+    }
+    Serial.print(bssid[i], HEX);
+    if (i != 5) {
+      Serial.print(":");
+    }
+    else {
+      Serial.println();
+    }
+  }
 
   // print the received signal strength:
-  long rssi = WiFi.RSSI();
-  Serial.print("signal strength (RSSI):");
+  int32_t rssi = WiFi.RSSI();
+  Serial.print("Signal strength (RSSI): ");
   Serial.println(rssi);
 
   // print the encryption type:
-  byte encryption = WiFi.encryptionType();
-  Serial.print("Encryption Type:");
-  Serial.println(encryption, HEX);
+  Serial.print("Encryption Type: ");
+  printEncryptionType(WiFi.encryptionType());
+  Serial.println();
+}
+
+void printEncryptionType(uint8_t encryptionType) {
+  // read the encryption type and print out the name:
+  switch (encryptionType) {
+    case ES_WIFI_SEC_OPEN:
+      Serial.println("OPEN");
+      break;
+    case ES_WIFI_SEC_WEP:
+      Serial.println("WEP");
+      break;
+    case ES_WIFI_SEC_WPA:
+      Serial.println("WPA");
+      break;
+    case ES_WIFI_SEC_WPA2:
+      Serial.println("WPA2");
+      break;
+    case ES_WIFI_SEC_WPA_WPA2:
+      Serial.println("WPA_WPA2");
+      break;
+    case ES_WIFI_SEC_WPA2_TKIP:
+      Serial.println("WPA_TKIP");
+      break;
+    case ES_WIFI_SEC_UNKNOWN:
```

Comment:
You  could do it like this to avoid duplicate `Serial.println`
```
+    case ES_WIFI_SEC_UNKNOWN:
+    default:
+      Serial.println("UNKNOWN");
+      break;
```

Reply:
Ok, I will add your suggestions in the next commit